src/UBAFeeCalculator/UBAFeeSpokeCalculatorAnalog.test.ts

```diff
@@ -0,0 +1,182 @@
+      const chainId = 1;
+      const flowType = "outflow";
+      const fee = getEventFee(amount, flowType, lastRunningBalance, lastIncentiveBalance, chainId, defaultConfig);
+      expect(fee.balancingFee.toString()).toEqual("0");
+    });
+
+    it("should return a balance fee 0 if lastIncentiveBalance is 0", () => {
+      const amount = BigNumber.from(10);
+      const lastRunningBalance = BigNumber.from(1000);
+      const lastIncentiveBalance = BigNumber.from(0);
+      const chainId = 1;
+      const flowType = "outflow";
+      const fee = getEventFee(amount, flowType, lastRunningBalance, lastIncentiveBalance, chainId, defaultConfig);
+      expect(fee.balancingFee.toString()).toEqual("0");
+    });
+
+    describe("Should correctly apply a reward multiplier", () => {
+      for (const fee of [-23, 0, 23]) {
+        const signToZero = fee === 0 ? "==" : fee > 0 ? ">" : "<";
+        it(`should return a balance fee a discounted balance fee if the fee is ${signToZero} 0`, () => {
+          const amount = BigNumber.from(10);
+          const lastRunningBalance = BigNumber.from(1000);
+          const lastIncentiveBalance = BigNumber.from(1000);
+          const chainId = 1;
+          const flowType = "outflow";
+
+          const originalFee = getEventFee(
+            amount,
+            flowType,
+            lastRunningBalance,
+            lastIncentiveBalance,
+            chainId,
+            defaultConfig
+          );
+
+          const config = new UBAConfig(
+            {
+              default: toBN(0),
+            },
+            {
+              default: [
+                [toBN(0), toBN(0)],
+                [toBN(0), parseEther("2")],
+              ],
+            },
+            {
+              default: {
+                upperBound: {},
+                lowerBound: {},
+              },
+            },
+            {
+              default: [[toBN(0), parseEther("1")]],
+            },
+            {},
+            {
+              [chainId]: parseEther(String(fee)),
+            }
+          );
+
+          const modifiedFee = getEventFee(amount, flowType, lastRunningBalance, lastIncentiveBalance, chainId, config);
+          expect(modifiedFee.balancingFee.toString()).toEqual(originalFee.balancingFee.mul(fee).toString());
+        });
+      }
+    });
+
+    it("should return a discounted balance fee if the lastIncentiveBalance is positive", () => {
+      const amount = BigNumber.from(10);
+      const lastRunningBalance = BigNumber.from(1000);
+      const lastIncentiveBalance = BigNumber.from(1000);
+      const chainId = 1;
+      const flowType = "inflow";
+      const fee = getEventFee(amount, flowType, lastRunningBalance, lastIncentiveBalance, chainId, defaultConfig);
+      expect(fee.balancingFee.toString()).toEqual("20");
+    });
+  });
+
+  for (const [flowName, flowType] of [
+    ["getDepositFee", "inflow"],
+    ["getRefundFee", "outflow"],
+  ]) {
+    describe(flowName, () => {
```

Comment:
I'm not entireley sure what this test is supposed to achieve can you elaborate?

Reply:
These are meant to guarantee that the `getDepositFee`/`getRefundFee` return the same as an adjacent call to `getEventFee`.

Added this comment:

```
  // The following tests are designed around ensuring that our `getEventFee` specific functions (getDepositFee and getRefundFee) are
  // calculating the same fee as the `getEventFee` function. This is to ensure that the `getEventFee` function is working as expected
  // as well as the `getDepositFee` and `getRefundFee` functions.
  // Based on their design, these calculation functions should always return the same fee as the `getEventFee` function.
```

Reply:
Done: https://github.com/across-protocol/sdk-v2/pull/342/commits/abbdafb07855a718155ff9306655a93c0c0d2b9b